doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
-      possible cases. The name of the equation is specified
-      by :token:`naming_intropattern` (see :tacn:`intros`),
-      in particular ``?`` can be used to let Coq generate a fresh name.
+* `n` is a :gdef:`dependent premise` of `forall n:nat, n + 0 = n` because
+  `n` appears both in the binder of the `forall` and in the quantified statement
+  `n + 0 = n`.
 
-   .. tacv:: destruct @term with @bindings
+* `A` and `B` are :gdef:`non-dependent premises <non-dependent premise>` of `A -> B -> C`
+  because they don't appear in a `forall` binder.
+  (Recall that `A -> B` is notation for the term `forall _ : A, B`; the wildcard
+  `_` can't be referred to in the quantified statement.)
 
-      This behaves like :n:`destruct @term` providing explicit instances for
-      the dependent premises of the type of :token:`term`.
+.. tacn:: destruct @induction_clause_list
 
-   .. tacv:: edestruct @term
-      :name: edestruct
+   .. insertprodn induction_clause_list as_or_and_ipat
 
-      This tactic behaves like :n:`destruct @term` except that it does not
-      fail if the instance of a dependent premises of the type
-      of :token:`term` is not inferable. Instead, the unresolved instances
-      are left as existential variables to be inferred later, in the same way
-      as :tacn:`eapply` does.
+   .. prodn::
+      induction_clause_list ::= {+, @induction_clause } {? using @one_term {? with @bindings } } {? @occurrences }
+      induction_clause ::= @destruction_arg {? @as_or_and_ipat } {? eqn : @naming_intropattern } {? @occurrences }
+      destruction_arg ::= @natural
+      | @one_term_with_bindings
+      as_or_and_ipat ::= as {| @or_and_intropattern | @ident }
+      | as @equality_intropattern
 
-   .. tacv:: destruct @term using @term {? with @bindings }
+   Generates a subgoal for each constructor of the inductive or co-inductive
+   type in the :n:`@one_term` in :n:`@destruction_arg`.  Unlike :tacn:`induction`,
+   no induction hypothesis is generated by :tacn:`destruct`.
 
-      This is synonym of :n:`induction @term using @term {? with @bindings }`.
+   The :n:`@one_term` may contain holes that are denoted by “_”. In this case,
+   the tactic checks that all subterms matching the pattern in the conclusion
+   and the hypotheses are compatible and performs case analysis using this subterm.
 
-   .. tacv:: destruct @term in @goal_occurrences
+   .. todo PR Need more explanation about the tactic does when destruction_arg is
+      a one_term that's not just an identifier'
 
-      This syntax is used for selecting which occurrences of :token:`term`
-      the case analysis has to be done on. The :n:`in @goal_occurrences`
-      clause is an occurrence clause whose syntax and behavior is described
-      in :ref:`occurrences sets <occurrencessets>`.
+   :n:`{+, @induction_clause }`
+     Giving multiple :n:`@induction_clause`\s is equivalent to applying :n:`destruct`
+     serially on each :n:`@induction_clause`.
```

Comment:
I think "chaining calls to `destruct` for each `@induction_clause`" would read better.

Reply:
I don't quite see it.  "chaining" is not precisely defined.  The dictionary doesn't have a verb definition similar to this (not that that's definitive).

Reply:
OK maybe it's a Frenchism. We often say "chaîner" for `;`.

Reply:
Not a Frenchism.  In English it comes from a chain as in 
![image](https://user-images.githubusercontent.com/396076/122256918-7d29a280-ce9d-11eb-8350-201d0bfd6d6f.png)

"chaining" is linking or sequencing multiple things, as if in a chain.

Reply:
My comment was not that "chaining" isn't used somewhat informally in this sense, it was that "chaining" is not precise.

In basic calculus: "The chain rule states that the derivative of f(g(x)) is f'(g(x))⋅g'(x)."

https://web.mit.edu/wwmath/calculus/differentiation/chain.html: "This rule is called the chain rule because we use it to take derivatives of composites of functions by chaining together their derivatives."  Clearly quite different from `destruct`!

"chaîner" appears to be closer to "linked", I think.

```
chaîner
verb
 to be chained to another data item
```

Reply:
>  Clearly quite different from destruct!

"chaining" is not about `destruct` ; IIUC, it's about the `;` in the fact that `destruct a, b` is the same as `destruct a; destruct b`.

Reply:
Also, `;` is composition of tactics in a way that's very similar to composition of functions, and this similarity can be made technically precise.  In this way the "chain" in the chain rule is about the same thing as the "chain" in "chaining tactic calls"

Reply:
> >  Clearly quite different from destruct!

> "chaining" is not about `destruct` ; IIUC, it's about the `;` in the fact that `destruct a, b` is the same as `destruct a; destruct b`.

I was referring to `destruct a, b` of course, I just didn't spell it out completely.

> this similarity can be made technically precise.  In this way the "chain" in the chain rule is about the same thing as the "chain" in "chaining tactic calls"

Maybe so, but it doesn't make sense to define that term here--it's simpler just to explain `destruct a,b` with a short phrase.  "Chaining" is used in 9 other places in the doc.  If we make it a defined term, the other uses need to examined to make sure the definition fits them.

Not obvious to me how you would define it in a useful way.

Fun as it is to bikeshed small details, I would like to move on.  @Zimmi48 if you want to say "chained" here then I'll do that, but I do think that's vague.

Reply:
I'm quite fine with the current state, even though I find the terminology "compound tactic" less clear than "chaining tactics" (none of them being clearly defined in the manual yet).

Reply:
I dropped "compound tactic" and added hyperlinks.